miden-lib/comment_format.md

```diff
@@ -0,0 +1,224 @@
+# Format guidebook
+
+A guidebook defining the format of documentation comments and regular comments for `masm` procedures.
+
+## General
+
+Entire procedure doc comment should be created using the `#!` pair of symbols as the commenting sign.
+
+Doc comment for a procedure should have these blocks:
+
+- Procedure description.
+- Inputs and outputs.
+- Description of the values used in the "Inputs and outputs" block (optional).
+- Panic block (optional).
+- Invocation hint (optional and will become redundant after the procedure annotations will be implemented).
+
+Each block should be separated from the others with a blank line.
+
+Example:
+
+```masm
+#! Sets the code of the account the transaction is being executed against.
+#!
+#! Inputs:  [CODE_COMMITMENT]
+#! Outputs: []
+#!
+#! Where:
+#! - CODE_COMMITMENT is the hash of the code to set.
+#!
+#! Panics if:
+#! - this procedure is executed on the account which type differs from the `basic mutable`.
+#!
+#! Invocation: exec
+```
```

Comment:
I think it would be great if this was a full example here that showcases most, or all possible cases, so it can serve as a great first reference to look at. And only if one can't figure out from that example what a comment should look like, we can use the reference level guide below.

So basically I would just extend the current example with 
- every possible value type (a felt, a word, a `_ptr`) in either `Inputs` or `Outputs`.
- `Inputs` specifying Operand Stack, Advice Stack and Advice map.